bigtop-packages/src/common/apex/do-component-build

```diff
@@ -26,10 +26,10 @@ command -v mvn > /dev/null 2>&1 || {
 }
 
 # Build apex
-mvn clean package -DskipTests "$@"
+mvn clean package -DskipTests -Dhadoop.version=$HADOOP_VERSION "$@"
 
 # Find all the dependencies of apex
-mvn dependency:copy-dependencies -DoutputDirectory=`pwd`/all-deps -DincludeScope=runtime -DexcludeScope=test
+mvn dependency:copy-dependencies -DoutputDirectory=`pwd`/all-deps -DincludeScope=runtime -DexcludeScope=test -Dhadoop.version=$HADOOP_VERSION
```

Comment:
@chinmaykolhatkar look at how the relevant jar files are filtered here: https://github.com/atrato/apex-cli-package/blob/master/pom.xml

It does not require the brittle exclusions list that you have below, because it works based on the apex-engine dependencies.